src/cascadia/TerminalControl/ControlCore.h

```diff
@@ -152,6 +152,10 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         bool HasSelection() const;
         bool CopyOnSelect() const;
         Windows::Foundation::Collections::IVector<winrt::hstring> SelectedText(bool trimTrailingWhitespace) const;
+        Core::Point SelectionAnchor() const;
+        Core::Point SelectionEnd() const;
+        bool MovingEnd() const;
+        bool MovingCursor() const;

```

Comment:
It looks like all four of these methods are called at the same time in the same caller. They're also _all related_.

* Each one requires the lock to be taken, just to read some POD: lock, unlock, four times per selection update
* Once the content process split happens, _each one_ is an RPC call (!): four times for every selection update, each of which could have high latency or contention

Should we combine them into one struct?